codelabs/theme_extension/theme_extension.md

```diff
@@ -0,0 +1,250 @@
+  'base': Blockly.Themes.Classic,
+});
+```
+
+And then add a line to inject this theme in function start() in `index.js`
+```js
+function start() {
+    // Create main workspace.
+    workspace = Blockly.inject('blocklyDiv',
+      {
+        toolbox: toolboxCategories,
+        theme: Blockly.Themes.Halloween,
+      });
+ } 
+```
+
+Note: At this point, we have just created a new theme. But it does not have any customizations. We will do that next.
+
+## Customize Components 
+
+Within the Halloween theme definition, you can customize the colours of multiple components:
+
+```js
+Blockly.Themes.Halloween = Blockly.Theme.defineTheme('halloween', {
+  'base': Blockly.Themes.Classic,
+  'componentStyles': {
+    'workspaceBackgroundColour': '#ff7518',
+    'toolboxBackgroundColour': '#F9C10E',
+    'toolboxForegroundColour': '#fff',
+    'flyoutBackgroundColour': '#252526',
+    'flyoutForegroundColour': '#ccc',
+    'flyoutOpacity': 1,
+    'scrollbarColour': '#ff0000',
+    'insertionMarkerColour': '#fff',
+    'insertionMarkerOpacity': 0.3,
+    'scrollbarOpacity': 0.4,
+    'cursorColour': '#d0d0d0',
+    'blackBackground': '#333'
+  }
+});
+```
+
+### Test it
+
+Reload your web page. You should see a themed workspace!
+
+
+![Fun with colours.](theme_components_workspace.png)
+
+## Customize Category Styles 
+A category style currently only holds a colour property. It is the colour of the category on the toolbox.
+This value can either be defined as a hex value or as a hue. Usually these colours should be the same as 
+the colourPrimary on the majority of blocks in the category making it easy for users to tell what blocks 
+belong in what category.
+
+Update the Theme definition to have the category styles as below.
+
+```js
+Blockly.Themes.Halloween = Blockly.Theme.defineTheme('halloween', {
+  'base': Blockly.Themes.Classic,
+  'categoryStyles': {
+  'list_category': {
+     'colour': "#4a148c"
+   },
+   'logic_category': {
+     'colour': "#8b4513",
+   }
+   'loop_category': {
+     'colour': "#85E21F",
+   }
+   'text_category': {
+     'colour': "#FE9B13",
+   }
+  },
+  'componentStyles': {
+    'workspaceBackgroundColour': '#ff7518',
+    'toolboxBackgroundColour': '#F9C10E',
+    'toolboxForegroundColour': '#fff',
+    'flyoutBackgroundColour': '#252526',
+    'flyoutForegroundColour': '#ccc',
+    'flyoutOpacity': 1,
+    'scrollbarColour': '#ff0000',
+    'insertionMarkerColour': '#fff',
+    'insertionMarkerOpacity': 0.3,
+    'scrollbarOpacity': 0.4,
+    'cursorColour': '#d0d0d0',
+    'blackBackground': '#333'
+  }
+});
+
+```
+
+### Test it
+
+Click on different blocks in the component and you should see the colours that you applied show up.
+
+![Customized Block colors on the components](customized_categories.png)
+
+
+## Customize Block Styles 
+A block style currently only holds three different colour properties. They are 'colourPrimary', 
+'colourSecondary' and 'colourTertiary'. 
```

Comment:
Add something like "For more information on block styles visit our themes documentation" and then add a link to: https://developers.google.com/blockly/guides/configure/web/themes#block_style